autogpt_platform/backend/backend/data/graph.py

```diff
@@ -275,6 +275,54 @@ def sanitize(name):
                         f"Node {block.name} #{node.id} required input missing: `{name}`"
                     )
 
+            # Get input schema properties and check dependencies
+            input_schema = block.input_schema.model_json_schema()
+            properties = input_schema.get("properties", {})
+            required_fields = set(input_schema.get("required", []))
+
+            # Validate dependencies between fields
+            for field_name, field_schema in properties.items():
+                if "depends_on" in field_schema:
+                    dependencies = field_schema["depends_on"]
+
+                    # Check if dependent field has value in input_default
+                    has_value = (
+                        field_name in node.input_default
+                        and node.input_default[field_name] is not None
+                        and str(node.input_default[field_name]).strip() != ""
+                    ) or (
+                        "default" in field_schema
+                        and field_schema["default"] is not None
+                    )
+                    must_have_value = field_name in required_fields
+
+                    # Check for missing dependencies when dependent field is present
+                    missing_deps = [
+                        dep
+                        for dep in dependencies
+                        if dep not in node.input_default
+                        or not node.input_default[dep]
+                        or str(node.input_default[dep]).strip() == ""
```

Comment:
This check seems replicated in 3 places, and some of them are missing the default ("default" field) value check.
We can add a lambda / inner function like:

```
            def has_value(name):
                return (
                    name in node.input_default and
                    node.input_default[name] is not None and
                    str(node.input_default[name]).strip() != ""
                ) or (
                    name in input_schema and
                    input_schema[name].default is not None
                )           
```